apps/web-staking/src/app/hooks/useGetRedemptions.ts

```diff
@@ -4,61 +4,79 @@ import { useAccount } from 'wagmi';
-        } catch (error) {
-            setRedemptionsLoaded(false);
-            setRedemptionsLoading(false);
-            console.error('Failed to load user redemptions:', error);
-        }
-    }, [address, chainId, setRedemptions, setRedemptionsLoaded]);
 
     const loadRedemptions = useCallback(async () => {
-        const now = Date.now();
-        if (now - lastLoadedRef.current < 1000) {
+        if (!address || !chainId || !isConnected) {
+            console.log("Cannot load redemptions - missing requirements", { address, chainId, isConnected });
             return;
         }
-        setRedemptionsLoading(true);
-        lastLoadedRef.current = now;
-        if (redemptionsLoaded) {
-            await updateUserRedemptions();
-        } else {
-            await loadAllUserRedemptions();
+        const now = Date.now();
+        // Do not reload if already loaded in the last second
+        if (now - lastLoadedRef.current > 1000) {
+            lastLoadedRef.current = now;
+
+            setRedemptionsLoading(true);
+            try {
+                // If address changed or not loaded yet, do a fresh load
+                if (loadedAddress !== address || !redemptionsLoaded) {
+                    const redemptionsFromChain = await getAllRedemptions(getNetwork(chainId), address);
+                    setRedemptions(redemptionsFromChain);
+                    setRedemptionsLoaded(true);
+                } else {
+                    const refreshedRedemptions = await refreshRedemptions(getNetwork(chainId), address, redemptions);
+                    setRedemptions(refreshedRedemptions);
+                }
+            } catch (error) {
+                console.error('Failed to load/update redemptions:', error);
+                setRedemptionsLoaded(false);
+                setRedemptions(INITIAL_REDEMPTIONS);
+            } finally {
+                setRedemptionsLoading(false);
+            }
         }
-        setRedemptionsLoading(false);
-    }, [redemptionsLoaded, updateUserRedemptions, loadAllUserRedemptions]);
+    }, [
+        address,
+        chainId,
+        isConnected,
+        loadedAddress,
+        redemptionsLoaded,
+        redemptions,
+        setRedemptions,
+        setRedemptionsLoaded,
+    ]);
 
     useEffect(() => {
-        if (address && chainId) {
-            loadRedemptions();
+        // Reset if disconnected or chain changed
+        if (!address || !chainId || !isConnected) {
+            setRedemptionsLoaded(false);
+            setRedemptions(INITIAL_REDEMPTIONS);
+            setLoadedAddress(undefined);
+            return;
         }
-    }, [address, chainId, loadRedemptions]);
+
+        // Handle address change
+        if (loadedAddress !== address) {
+            setRedemptionsLoaded(false);
+            setRedemptions(INITIAL_REDEMPTIONS);
+            setLoadedAddress(address);
+        }
+
+        loadRedemptions();
+    }, [address, chainId, isConnected, loadedAddress]);
```

Comment:
We should not set a dependency inside the useEffect, having `loadedAddress` in the deps will make this rerender multiple times since we will always call `setLoadedAddress()` within the useEffect.

Reply:
This has been removed, The loadedAddress has been removed completely per your other suggestion.